### ISSUE TITLE
can not converge if hidden_dim of transformer is setted to 512 （the default is 256） 

### ISSUE THREAD

**Issue by YoungSun3Pan** [facebookresearch/detr#314]
1. I'm trying to run training DETR with hidden_dim of transformer as 512 and getting an error
2. I started with default command, and it work well: `python -m torch.distributed.launch --nproc_per_node=4 --use_env main.py --hidden_dim=256`
3. But DETR can not converge if hidden_dim is setted to 512:`python -m torch.distributed.launch --nproc_per_node=4 --use_env main.py --hidden_dim=512`

Could you, please, help me with this problem? Any advice regarding optimal parameters are appreciated!

**Comment by YoungSun3Pan** [comment]
map is always 0 and train loss have not decreased。

**Comment by alcinos** [comment]
I'm not sure exactly how you are training your model or on which dataset, but if you are training on one node only using the command you posted, you'll have a total batch-size of 2*4=8 which is probably too small for reliable convergence. You may try to either increase the batch-size or decrease the learning rate.
Best of luck
